lib/whitehall_importer/integrity_check_error.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module WhitehallImporter
+  class IntegrityCheckError < AbortImportError
+    attr_reader :problems, :payload
+
+    def initialize(state, problems, payload)
+      @problems = problems
+      @payload = payload
+      super("#{state} Integrity check failed")
```

Comment:
This might read slightly weird if it's like "draft Integrity check failed" may want to adjust capitalisation.

Reply:
Yeah, that was going to be my recommendation too if I didn't have time to finish the harder stuff today..